stablehlo/reference/Element.cpp

```diff
@@ -350,6 +350,33 @@ Element min(const Element &e1, const Element &e2) {
       });
 }
 
+// These cases are implementation-defined, and the behavior of this interpreter:
+// For integers: if rhs is 0, the result is the same as `lhs`.
+//   For signed integers, if lhs is INT_MAX and rhs is -1, the result is 0.
+// For floats: use APFloat.mod(). If rhs is 0, the result is 0x7FF8000000000000
```

Comment:
Not sure if we should add comment about what behavior we choose for domain errors, given that we are not doing this religiously for other ops like `add`. 
 
https://github.com/llvm/llvm-project/blob/b309bc04eebc9c736b6c34d73d520a6ef7baf302/llvm/lib/Support/APFloat.cpp#L2167 seems to be equivanet to std::fmod

and `std::fmod` already calls out that the domain errors are implementation defined. 

IMO, any behavior is fine for us to have. Having said that maybe we can remove the special case for "divide by zero". As we know that we added it for parity with HLO, but having parity with HLO on cases which are free to vary across  implementations, does not seem worth following unless we have a definitive use-case in mind (a naive  use case could be minimizing false positives while  cross validating results with hlo interpreter)  

We can seek opinion from @burmako on this.

Reply:
Add @burmako as reviewer for more comments on this.

Reply:
Let's drop the comment. G5 in https://github.com/openxla/stablehlo/blob/main/docs/reference.md#testing-guidelines explicitly discourages writing tests for implementation-defined behavior, so I think we should extrapolate this to documenting implementation-defined behavior.

Reply:
Agreed with you guys, removed.